05/app.js

```diff
@@ -1,53 +1,85 @@
 const stats = {
     paragraphs: {
-        'p1': 0,
+        p1: 0
     },
     links: {
-        '/dolor.html': 0,
+        '/dolor.html': 0
     }
 };
 
 /* tutaj umieść swój kod */
+// mozna to rowniez ogarnac za pomoca dwoch funkcji
+function calculateClicks(element, type) {
+    element.addEventListener('click', function (e) {
+        if (type === 'paragraphs' && e.target.tagName === 'P') {
+            const id = element.dataset.id;
 
+            if (stats[type][id]) {
+                stats[type][id]++;
+            } else {
+                stats[type][id] = 1;
+            }
```

Comment:
Ta część jest prawie identyczna z 26-31. Tutaj spokojnie można napisać funkcję, która nie będzie powtarzać tego elementu. Wystarczy, że w tej funkcję będę 2 parametry tj. elementType, elementId

Reply:
Jasne, dobry pomysł :) można by jeszcze pomyśleć o innych skrótach, ale pytanie czy będzie to zrozumiałe. Tutaj musimy dodawać właściwości do obiektu, co samo w sobie może być średnio zrozumiałe, dlatego nie chciałem zaciemniać kodu.